apiserver/apiserver.go

```diff
@@ -522,79 +559,72 @@ func (srv *Server) mongoPinger() error {
 	}
 }
 
-// changeCertListener wraps a TLS net.Listener.
-// It allows connection handshakes to be
-// blocked while the TLS certificate is updated.
-type changeCertListener struct {
-	net.Listener
-
-	// srv holds the server that the listener was created by.
-	srv *Server
-
-	// certChanged is used to pass in new certificate information.
-	certChanged <-chan params.StateServingInfo
-
-	// mu guards the fields below it.
-	mu sync.Mutex
-
-	// cert holds the current certificate used for tls.Config
-	cert tls.Certificate
-}
-
-func (srv *Server) newChangeCertListener(lis net.Listener, certChanged <-chan params.StateServingInfo, cert tls.Certificate) *changeCertListener {
-	return &changeCertListener{
-		Listener:    lis,
-		srv:         srv,
-		cert:        cert,
-		certChanged: certChanged,
+// getCertificate returns the certificate appropriate for a client
+// that's just introduced themselves.
+func (srv *Server) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	logger.Infof("getCertificate - clientHello.ServerName: %q", clientHello.ServerName)
+	cert, shouldUse := srv.localCertificate(clientHello.ServerName)
+	if shouldUse {
```

Comment:
It might be clearer if we just did:

if !shouldUse {
  logger.Log()
}
return cert, nil

Rather than making it look like we do something different based on shouldUse, just make it clear that all we're doing is adding logging.